cloudbuild/build.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+set -eo pipefail
+
+env
+
+# get our service key file out of secret manager
+gcloud secrets versions access latest --secret=devstack-service-key --project="$PROJECT_ID" > /workspace/devstack.json
+
+# get our SSH key
+mkdir /root/.ssh
+chmod 700 /root/.ssh
+gcloud secrets versions access latest --secret=cloudbuild-ssh-key --project="$PROJECT_ID" > /root/.ssh/id_ed25519
+chmod 600 /root/.ssh/id_ed25519
+
+# sultan expects sudo to be installed
+apt-get update
+apt-get install -y sudo
+
+export USER=cloudbuild
+export HOME=/root
+export TERM=dumb  # make tput shut up
+
+# cloudbuild environment requires some trickiness
+mkdir /tmp/.ansible
+export ANSIBLE_SSH_ARGS="-C -o ControlMaster=auto -o ControlPersist=60s -o 'ControlPath=/tmp/.ansible/ansible-ssh-%h-%p-%r'"
+
+# set up requirements
+pip install -r requirements.txt
+
+# use our config
+cp cloudbuild/configs."$CONFIG" configs/.configs.cloudbuild
+
+echo "CONFIG DEBUG:"
```

Comment:
I suggest echoing a line on top of these echo messages like `------` for a more efficient skimming, but that's not necessary.